marine/marine.py

```diff
@@ -258,3 +263,16 @@ def _expand_macros(
             ret_value = tuple(zip(*expanded_with_indices))
             self._macros_cache[macro_key] = ret_value
             return ret_value
+
+    def _detect_encap(self, fields: List[str]) -> int:
+        encap_key = frozenset(fields)
+        if encap_key in self._encap_cache:
+            return self._encap_cache[encap_key]
+
+        fields_protocols = frozenset(field.split(".")[0].lower() for field in fields)
+        if fields_protocols.intersection(self.WIFI_RADIO_PROTOCOLS):
+            self._encap_cache[encap_key] = encap_consts.ENCAP_IEEE_802_11_RADIOTAP
+            return encap_consts.ENCAP_IEEE_802_11_RADIOTAP
+
+        self._encap_cache[encap_key] = encap_consts.ENCAP_ETHERNET
+        return encap_consts.ENCAP_ETHERNET
```

Comment:
I think it will look nicer if you do the `if`s to decide what to put inside `self._encap_cache` and then simply return `self._encap_cache[encap_key]`